dotCMS/src/main/java/com/dotcms/api/APIProvider.java

```diff
@@ -0,0 +1,155 @@
+            final HTMLPageAssetAPI htmlPageAssetAPI,
+            final CategoryAPI categoryAPI,
+            final ContentHelper contentHelper,
+            final PermissionAPI permissionAPI) {
+        this.identifierAPI = identifierAPI;
+        this.hostAPI = hostAPI;
+        this.languageAPI = languageAPI;
+        this.fileAssetAPI = fileAssetAPI;
+        this.versionableAPI = versionableAPI;
+        this.userAPI = userAPI;
+        this.contentletAPI = contentletAPI;
+        this.htmlPageAssetAPI = htmlPageAssetAPI;
+        this.categoryAPI = categoryAPI;
+        this.contentHelper = contentHelper;
+        this.permissionAPI = permissionAPI;
+    }
+
+    public static class Builder {
+
+        private IdentifierAPI identifierAPI = APILocator.getIdentifierAPI();
+        private HostAPI hostAPI = APILocator.getHostAPI();
+        private LanguageAPI languageAPI = APILocator.getLanguageAPI();
+        private FileAssetAPI fileAssetAPI = APILocator.getFileAssetAPI();
+        private VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+        private UserAPI userAPI = APILocator.getUserAPI();
+        private ContentletAPI contentletAPI = APILocator.getContentletAPI();
+        private HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+        private CategoryAPI categoryAPI = APILocator.getCategoryAPI();
+        private ContentHelper contentHelper = ContentHelper.getInstance();
+        private PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+        public Builder withIdentifierAPI(IdentifierAPI identifierAPI) {
+            this.identifierAPI = identifierAPI;
+            return this;
+        }
+
+        public Builder withHostAPI(HostAPI hostAPI) {
+            this.hostAPI = hostAPI;
+            return this;
+        }
+
+        public Builder withLanguageAPI(LanguageAPI languageAPI) {
+            this.languageAPI = languageAPI;
+            return this;
+        }
+
+        public Builder withFileAssetAPI(FileAssetAPI fileAssetAPI) {
+            this.fileAssetAPI = fileAssetAPI;
+            return this;
+        }
+
+        public Builder withVersionableAPI(VersionableAPI versionableAPI) {
+            this.versionableAPI = versionableAPI;
+            return this;
+        }
+
+        public Builder withUserAPI(UserAPI userAPI) {
+            this.userAPI = userAPI;
+            return this;
+        }
+
+        public Builder withContentletAPI(ContentletAPI contentletAPI) {
+            this.contentletAPI = contentletAPI;
+            return this;
+        }
+
+        public Builder withHtmlPageAssetAPI(HTMLPageAssetAPI htmlPageAssetAPI) {
+            this.htmlPageAssetAPI = htmlPageAssetAPI;
+            return this;
+        }
+
+        public Builder withCategoryAPI(CategoryAPI categoryAPI) {
+            this.categoryAPI = categoryAPI;
+            return this;
+        }
+
+        public Builder withContentHelper(ContentHelper contentHelper) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'contentHelper' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5685344)